[FEAT] Add doc to simplify podspecs for Native Modules and Native Components (#3321)
facebook/react-native-website: docs/the-new-architecture/backward-compatibility-turbomodules.md

The Apple platform installs Turbo Native Modules using [Cocoapods](https://cocoapods.org) as a dependency manager.

> If you are already using the [`install_module_dependencies`](https://github.com/facebook/react-native/blob/82e9c6ad611f1fb816de056ff031716f8cb24b4e/scripts/react_native_pods.rb#L145) function, **there is nothing to do**. The function already takes care of installing the proper dependencies when the New Architecture is enabled and avoid them when it is not enabled.

Reviewer: ```suggestion
If you are already using the [`install_module_dependencies`](https://github.com/facebook/react-native/blob/82e9c6ad611f1fb816de056ff031716f8cb24b4e/scripts/react_native_pods.rb#L145) function, then **there is nothing to do**. The function already takes care of installing the proper dependencies when the New Architecture is enabled and avoids them when it is not enabled.
```

end
```

> You want to install the extra dependencies when the New Architecture is enabled and you want to avoid installing them when it's not.

Reviewer: ```suggestion
You should install the extra dependencies when the New Architecture is enabled, and avoid installing them when it's not.
```
